build_all.bat

```diff
@@ -1,20 +1,34 @@
 @echo off
-rem build development version of the compiler; can be rerun safely
-rem TODO: call nimBuildCsourcesIfNeeded or auto-generate this file (from a nim script)
-rem to avoid duplication.
-if not exist csources_v1 (
-  git clone --depth 1 https://github.com/nim-lang/csources_v1.git
+rem Build development version of the compiler; can be rerun safely
+rem bare bones version of ci/funs.sh adapted for windows.
+
+rem Read in some common shared variables (shared with other tools),
+rem see https://stackoverflow.com/questions/3068929/how-to-read-file-contents-into-a-variable-in-a-batch-file
+for /f "delims== tokens=1,2" %%G in (config/build_config.txt) do set %%G=%%H
+SET nim_csources=bin\nim_csources_%nim_csourcesHash%.exe
+echo "building from csources: %nim_csources%"
+
+if not exist %nim_csourcesDir% (
+  git clone -q --depth 1 %nim_csourcesUrl% %nim_csourcesDir%
 )
-if not exist bin\nim.exe (
-  cd csources_v1
-  git checkout a8a5241f9475099c823cfe1a5e0ca4022ac201ff
-  if PROCESSOR_ARCHITECTURE == AMD64 (
+
+if not exist %nim_csources% (
+  cd %nim_csourcesDir%
+  git checkout %nim_csourcesHash%
+  echo "%PROCESSOR_ARCHITECTURE%"
+  if "%PROCESSOR_ARCHITECTURE%"=="AMD64" (
     SET ARCH=64
   )
   CALL build.bat
   cd ..
+  cp bin\nim.exe  %nim_csources%
 )
-bin\nim.exe c --skipUserCfg --skipParentCfg koch
-koch.exe boot -d:release --skipUserCfg --skipParentCfg
-koch.exe tools --skipUserCfg --skipParentCfg
 
+if "%nim_build_all_only_csources%"=="1" (
```

Comment:
Why? Is anybody supposed to remember that even the "build_**ALL**.bat" file has options and configuration and command line arguments and environment variables?

Reply:
this is to allow reusing the part that builds nim_csources in both CI (so it gets tested against regressions) and build_all.bat

I'm trying to keep this as simple as possible; with posix, `ci/funs.sh` is great because you can `. ci/funs.sh` and use any of the APIs it provides; but with cmd.exe, i'm not sure what's the equivalent.

2 options:
* treat `nim_build_all_only_csources` as internal use only (used by CI as done in this PR), maybe with some comments explaning this
* split build_all.bat in 2: `ci/funs.bat` (used by CI) and build_all.bat (which calls ci/funs.bat and then builds koch, boot, tools)

Reply:
Since we already generate the CI configuration, why not generate these scripts (including the Bash scripts) too...

Reply:
done, it removes a little bit of duplication